tool/tctl/common/semaphore_command.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gravitational/kingpin"
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/asciitable"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/service"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/trace"
+)
+
+// SemaphoreCommand implements basic semaphore operations.
+type SemaphoreCommand struct {
+	config *service.Config
+
+	username string
+	leaseID  string
+
+	format string
+
+	force   bool
+	verbose bool
+
+	scList   *kingpin.CmdClause
+	scDelete *kingpin.CmdClause
+}
+
+// Initialize allows SemaphoreCommand to plug itself into the CLI parser
+func (c *SemaphoreCommand) Initialize(app *kingpin.Application, config *service.Config) {
+	c.config = config
+
+	sc := app.Command("sessctl", "Session-control introspection & management").Alias("session-control")
+
+	c.scList = sc.Command("ls", "List active session-control leases").Alias("list")
+	c.scList.Flag("format", "Output format, 'text' or 'json'").Default(teleport.Text).StringVar(&c.format)
+	c.scList.Flag("verbose", "Output verbose lease details").Short('v').BoolVar(&c.verbose)
+	c.scList.Flag("user", "Filter output by username").StringVar(&c.username)
+
+	c.scDelete = sc.Command("rm", "Delete active session-control leases").Alias("delete").Hidden()
+	c.scDelete.Flag("force", "Permits unconstrained deletions").Short('f').BoolVar(&c.force)
```

Comment:
I wonder if `--force` should be a hidden flag as it permits the user to do dangerous things? We (as support) could always tell them to do it when there's no alternative but maybe we want to prevent people from discovering it.